homeassistant/components/duke_energy/config_flow.py

```diff
@@ -0,0 +1,65 @@
+"""Config flow for Duke Energy integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aiodukeenergy import DukeEnergy
+from aiohttp import ClientError, ClientResponseError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class DukeEnergyConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Duke Energy."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            api = DukeEnergy(
+                user_input[CONF_USERNAME], user_input[CONF_PASSWORD], session
+            )
+            try:
+                auth = await api.authenticate()
+            except ClientResponseError as e:
+                errors["base"] = "invalid_auth" if e.status == 404 else "cannot_connect"
+            except (ClientError, TimeoutError):
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                email = auth["email"].lower()
+                username = auth["cdp_internal_user_id"].lower()
```

Comment:
`internal_user_id` sounds quite unique to me tbh

Reply:
This is how the data comes back in their auth API response. I've got no control over naming here 🤷‍♂️

Reply:
No I mean, this sounds like a plausible unique_id to use

Reply:
Oh, so instead of the data check, you are suggesting to use:

```
await self.async_set_unique_id(device_unique_id)
self._abort_if_unique_id_configured()
```

correct?

Reply:
I think so, yes